src/main.rs

```diff
@@ -57,11 +212,12 @@ enum ExecuteError {
     RowRead(TryFromSliceError),
     Write(std::io::Error),
     TableFull,
+    Table(TableError),
 }
 
 #[derive(Debug)]
-enum ReplResult<'a> {
-    Rows(Vec<Row<'a>>),
+enum ReplResult {
+    Rows(Vec<OwnedRow>),
```

Comment:
I needed to take ownership of the rows instead of borrowing them through the table since that made getting the table rows in a loop impossible since Rust thinks the rows we are referencing that we pushed into the return vec are going to be referenced elsewhere while we then go and still have a mutable reference to the table that we borrowed from. That table mut ref could potentially change the data and invalidate the reference we put in the result vec.

Reply:
This only applies when we are selecting everything since we need to go through every page and get the rows.